frontend/src/components/upload/Dropzone.tsx

```diff
@@ -0,0 +1,180 @@
+import React, { useCallback, useRef, useState } from 'react'
+import { ACCEPTED_MIME_TYPES, MAX_FILE_SIZE_BYTES } from '../../config/upload'
+
+interface DropzoneProps {
+  onFileSelected: (file: File) => void
+  disabled?: boolean
+  accept?: string[]
+  maxSizeBytes?: number
+  multiple?: boolean
+  className?: string
+}
+
+function normalizeAcceptList(list: string[] | undefined): string[] {
+  if (!list || list.length === 0) return []
+  return Array.from(new Set(list.map((s) => s.trim()).filter(Boolean)))
+}
+
+function getExt(name: string): string {
```

Comment:
```suggestion
function getExtension(name: string): string {
```

No need to make this shorter it just makes this one more unreadable

Reply:
Corrected in penult commit